resources/vocabulary.md

```diff
@@ -2,5 +2,115 @@
 
 ## "Basic" concepts
 
+### Things that most programming languages can do
+
+#### Arithmetic operations
+This involves typing expressions like `1 + 2` and getting the answer `3` as an output. The operators `+`, `-`, `*` and `/` are common (universal?), while programming languages also tend to have other operators like `**` (exponential power, e.g. `2 ** 3` returns `8`) or `%` (modulo, e.g. `14 % 4` returns `2`).
+
+#### 'Simple' data types
+These are different kinds of values that can be used while programming. Languages have different types of values, but most have a variety of the following:
+- Numerical types
+  - `integer`, `long` - represent whole numbers, like `1` or `324`
+  - `float`, `double` - represent decimal numbers, like `3.14`
+- `character` - represent single characters, like `a` or `$`
+- `boolean` - these include the values `true` and `false`
+
+#### Booleans and comparison
+Programming languages use the concept of `true` and `false`, which are called 'boolean' values. Among other uses, these are important for comparing values with other values, using operators like `>`, `<`, `>=`, `<=`, or `==`. Using these operators, we can type expressions like `1 + 2 > 4` (answer: `false`) or `(2 ** 3 == 8) == true` (answer: `true`).  
+
+#### Variable assignment
+This means assigning a value to a variable and using the variable later on. For example:
+```
+variable = 3 // We assign the value 3 to 'variable'
+return variable + 4 // We add 4 to 'variable' - the computer will interpret 'variable' to be 3 and return 7
+```
+
+Variables can assign all kinds of things - integers (`3` in the above example), strings, objects, functions, and others.
+
+##### 'Composite' data types
+These are things that can contain other values. 
+- `array`
```

Comment:
I think I've only ever heard list and array, but would be good to mention both of those